tests/main/user-session-env/task.yaml

```diff
@@ -0,0 +1,63 @@
+summary: verify that user environment settings are added
+
+description: |
+    User environment variables are added via /etc/profile.d/snapd.sh (bash/sh
+    specific) or via /lib/environment.d/ helpers. Make sure that at least one of
+    the mechanisms works and XDG_DATA_DIRS and PATH are updated accordingly
+    inside the user session, no matter the shell they use.
+
+# environment generators were introduced in systemd 233, anything before that
+# will not work
+systems:
+   # no user sessions managed by systemd
+   - -ubuntu-14.04-*
+   # too old systemd, before user@.service template was introduced
+   - -amazon-linux-*
+   - -centos-7-*
+   - -debian-9-*
+   # too old systemd, before environment generators were introduced
+   - -ubuntu-16.04-*
```

Comment:
Don't we modify the distro so that these envs are set correctly? I.e. would it make sense to still run the test to ensure our distro modifications are there?

Reply:
We ship a `/etc/profile.d/snapd.sh`, but that's outside of user session. I can add a code path for distros that do not support user sessions too.

Reply:
This is done now.